controllers/nutanixmachine_controller.go

```diff
@@ -844,6 +844,12 @@ func (r *NutanixMachineReconciler) getMachineCategoryIdentifiers(rctx *nctx.Mach
 			categoryIdentifiers = append(categoryIdentifiers, &additionalCat)
 		}
 	}
+
+	_, err := GetOrCreateCategories(rctx.Context, rctx.NutanixClient, categoryIdentifiers)
```

Comment:
Can we move this section before the categories loop? This will prevent the additional categories to be created if they don't exist.

Reply:
updated